compaction/compaction_manager.cc

```diff
@@ -592,18 +603,24 @@ future<> compaction_manager::perform_major_compaction(table_state& t, std::optio
 namespace compaction {
 
 class custom_compaction_task_executor : public compaction_task_executor, public compaction_task_impl {
-    noncopyable_function<future<>(sstables::compaction_data&)> _job;
+    noncopyable_function<future<>(sstables::compaction_data&, sstables::compaction_read_monitor_generator_wrapper&)> _job;
 
 public:
-    custom_compaction_task_executor(compaction_manager& mgr, throw_if_stopping do_throw_if_stopping, table_state* t, tasks::task_id parent_id, sstables::compaction_type type, sstring desc, noncopyable_function<future<>(sstables::compaction_data&)> job)
+    custom_compaction_task_executor(compaction_manager& mgr, throw_if_stopping do_throw_if_stopping, table_state* t, tasks::task_id parent_id, sstables::compaction_type type, sstring desc, noncopyable_function<future<>(sstables::compaction_data&, sstables::compaction_read_monitor_generator_wrapper&)> job)
         : compaction_task_executor(mgr, do_throw_if_stopping, t, type, std::move(desc))
         , compaction_task_impl(mgr._task_manager_module, tasks::task_id::create_random_id(), 0, "compaction group", t->schema()->ks_name(), t->schema()->cf_name(), "", parent_id)
         , _job(std::move(job))
-    {}
+    {
+        _status.progress_units = "bytes";
+    }
```

Comment:
cannot we set it only once in the base class?

Reply:
progress unit for compaction should only be in bytes right?

Reply:
For executors it is always in bytes, but for other compaction tasks (not included in this patch) not necessarily. It may not be the best idea to sum all the sstables sizes from each table on each shard to find `progress.completed` for top level compaction task especially since it may change.

It was planned to count a progress of upper level tasks as percentage of work performed by children where each child contributes equally to the percentage no matter its job size. That's a little out of scope for this patch though.